src/components/TokenSelectorComponents.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { ComponentType, useRef, useEffect, useCallback } from 'react'
+import { components, MenuListComponentProps } from 'react-select'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faSearch } from '@fortawesome/free-solid-svg-icons'
+
+// Inspired by https://github.com/JedWatson/react-select/issues/3111#issuecomment-470911304
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export const MenuList: ComponentType<MenuListComponentProps<any>> = props => {
+  const {
+    selectProps: { onInputChange, onMenuInputFocus, inputValue, value },
+    setValue,
+  } = props
+
+  const ariaAttributes = {
+    // 'aria-autocomplete': 'list',
+    'aria-label': props.selectProps['aria-label'],
+    'aria-labelledby': props.selectProps['aria-labelledby'],
+  }
+
+  // Since I couldn't for the life of me figure out a way to call some sort of close on the menu,
+  // David offered the idea to instead set a value on the select.
+  // When a value is selected, the menu is closed. The value is whatever was selected already.
+  const onCloseButtonClick = useCallback((): void => setValue(value, 'set-value'), [setValue, value])
+
+  // On input's change event, we call the select's `onInputChange` handler with the appropriated action
+  const onChange = useCallback(
+    (e): void => {
+      onInputChange &&
+        onInputChange(e.currentTarget.value, {
+          action: 'input-change',
+        })
+    },
+    [onInputChange],
+  )
+
+  // To avoid triggering side effects on upper layers that causes the menu to be closed,
+  // we stopPropagation and trigger a focus
+  const onInputClick = useCallback((e): void => {
+    e.stopPropagation()
+    e.currentTarget.focus()
+  }, [])
+
+  // for autofocus
+  const inputRef = useRef<HTMLInputElement>(null)
+  useEffect(() => inputRef.current?.focus(), [])
+
+  return (
+    // <components.MenuList {...props}>
+    <>
+      <div className="header" style={{ display: 'flex', justifyContent: 'space-between' }}>
+        <h2>Select token</h2>
+        <button type="button" onClick={onCloseButtonClick}>
+          X
+        </button>
+      </div>
+      <div className="searchContainer" style={{ display: 'flex' }}>
+        <FontAwesomeIcon icon={faSearch} />
+        <input
+          ref={inputRef}
+          style={{
+            width: '100%',
+            boxSizing: 'border-box',
+          }}
+          autoCorrect="off"
+          autoComplete="off"
+          spellCheck="false"
+          type="text"
+          placeholder="Search by token Name, Symbol or Address"
```

Comment:
i dont think Name/Symbol/... should be capitalised

Reply:
(uber nit hehe)

Reply:
I got this from the design: https://user-images.githubusercontent.com/31534717/75153817-963c9c00-570c-11ea-95b1-4a40f362eceb.png

(Don't tell anyone, but actually, search by address is not implemented